code/modules/clothing/head/jobs.dm

```diff
@@ -235,38 +237,59 @@
 	become_hearing_sensitive(ROUNDSTART_TRAIT)
 	QDEL_NULL(atom_storage)
 
+/obj/item/clothing/head/fedora/inspector_hat/proc/set_prefix(desired_prefix)
+
+	prefix = desired_prefix
+
+	// Regenerated the phrases here.
+	for(var/old_regex in phrases_by_regex)
+		var/old_phrase = phrases_by_regex[old_regex]
+		var/obj/item/old_item = items_by_regex[old_regex]
+		items_by_regex -= old_regex
+		phrases_by_regex -= old_regex
+		set_phrase(old_phrase,old_item)
+
+	return TRUE
+
+/obj/item/clothing/head/fedora/inspector_hat/proc/set_phrase(desired_phrase,obj/item/associated_item)
+
+	var/regex/phrase_regex = regex("[prefix]\[\\s\\W\]+[desired_phrase]","i")
```

Comment:
Silly question but if I put in some random regex as a key, like `"Go go gadget [a-z]{4}`... will it... work?

Reply:
If you try typing in your own regex as a prefix or as a phrase, it will not work thanks to how byond handles the regex proc (it's why we gotta put an extra \ in front of everything).